### PR TITLE
Error message

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: include/aos/common/tools/log.hpp [hunk 1]
```diff
@@ -195,14 +195,9 @@ class Log : private NonCopyable {
...
     {
-        *this << err.Message();
-
-        if (err.Errno() != 0) {
-            *this << " [" << err.Errno() << "]";
-        }
-
-        if (err.FileName() != nullptr) {
-            *this << " (" << err.FileName() << ":" << err.LineNumber() << ")";
+        StaticString<AOS_CONFIG_ERROR_PRINT_MESSAGE_LEN> tmpStr;
```
Comment (al1img) [inline include/aos/common/tools/log.hpp:198]: Add empty line after var definition.

File: include/aos/common/tools/error.hpp [hunk 2]
```diff
@@ -261,6 +293,7 @@ class Error {
     int         mErrno;
     const char* mFileName;
     int         mLineNumber;
+    char        mMessage[AOS_CONFIG_ERROR_MESSAGE_LEN] {};
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply include/aos/common/tools/error.hpp:296]: Done

File: include/aos/common/config.hpp [hunk 3]
```diff
@@ -634,4 +634,18 @@
...
+ */
+#ifndef AOS_CONFIG_ERROR_MESSAGE_LEN
+#define AOS_CONFIG_ERROR_MESSAGE_LEN 64
+#endif
+
+/**
+ * Max error print message len.
+ */
+#ifndef AOS_CONFIG_ERROR_PRINT_MESSAGE_LEN
+#define AOS_CONFIG_ERROR_PRINT_MESSAGE_LEN 128
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply include/aos/common/config.hpp:648]: Done

File: include/aos/common/tools/error.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply include/aos/common/tools/error.hpp:296]: Done

File: include/aos/common/tools/error.hpp [hunk 4]
```diff
@@ -231,6 +251,18 @@ class Error {
...
 private:
+    /**
+     * Copies error message.
+     *
+     * @param msg error message.
+     */
+    void CopyMessage(const char* msg)
+    {
+        if (msg != nullptr) {
+            strncpy(mMessage, msg, sizeof(mMessage) - 1);
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply include/aos/common/tools/error.hpp:262]: Done

File: include/aos/common/tools/log.hpp [hunk 5]
```diff
@@ -195,20 +195,19 @@ class Log : private NonCopyable {
...
+        if (tmpStr.Convert(err).IsNone()) {
+            return *this << tmpStr;
         }
 
         return *this;
     }
 
 private:
+    // AOS_CONFIG_ERROR_STR_LEN
+    constexpr static auto cMaxErrorStrLen = AOS_CONFIG_ERROR_STR_LEN;
```
Comment (mykola-kobets-epam) [inline include/aos/common/tools/log.hpp:209]: please move cMaxErrorStrLen to  string header.

**Review by mykola-kobets-epam — APPROVED** [review]

File: include/aos/common/tools/log.hpp (hunk 5 above)
Comment (MykolaSolyanko) [reply include/aos/common/tools/log.hpp]: Done. move to error.hpp

**Review by al1img — APPROVED** [review]

**Pull request merged**